Dockerfile.dev

```diff
@@ -8,14 +8,21 @@ FROM python:2.7-slim
 MAINTAINER bhearsum@mozilla.com
 
 # Some versions of the python:2.7 Docker image remove libpcre3, which uwsgi needs for routing support to be enabled.
-# Node and npm are to build the frontend. nodejs-legacy is needed by this version of npm.
 # mysql-client is needed to import sample data
 # libmysqlclient-dev is required to use SQLAlchemy with MySQL, which we do in production.
 # libfontconfig1 is required by phantomjs
 # git is needed to send coverage information
+# gcc is needed to build Python SQL module as well as uwsgi.
+# Use nodesource repo to get more modern nodejs. Node and npm are to build the frontend.
 RUN apt-get -q update \
-    && apt-get -q --yes install netcat libpcre3 libpcre3-dev nodejs nodejs-legacy npm libmysqlclient-dev mysql-client \
-                                libfontconfig1 netcat git curl \
+    && apt-get -q --yes install apt-transport-https curl \
+    && curl -sLf -o /dev/null 'https://deb.nodesource.com/node_4.x/dists/jessie/Release' \
+    && curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | apt-key add - \
+    && echo 'deb https://deb.nodesource.com/node_4.x jessie main' > /etc/apt/sources.list.d/nodesource.list \
+    && apt-get update \
+    && apt-get -q --yes install nodejs gcc \
+                       netcat libpcre3 libpcre3-dev libmysqlclient-dev mysql-client \
+                       libfontconfig1 netcat git \
     && apt-get clean
```

Comment:
These changes will need to get carried over to "Dockerfile" as well (that's the one we use to build the production images). We should probably explicitly test that image as well, to make sure it can still build the app.